### PR TITLE
My updates

### PR DESCRIPTION
**Pull request by hvlxh**
Better, fixed types.
use any case (pascel, camel) in client.write or client.queue

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -181,30 +177,43 @@ declare module "bedrock-protocol" {
...
-  }
+    gamemodeId: number
+    portV4: number
+    portV6: number
+    version: string
+    protocol: string
+
+    constructor(obj: object, port: number, version: string)
 
-  export interface RealmsOptions {
```
Comment (LucienHH) [inline index.d.ts:200]: Why remove this?

File: index.d.ts [hunk 2]
```diff
@@ -1,173 +1,169 @@
...
-      motd: string,
-      // The sub-header for the MOTD shown in the server list.
+      motd: string
       levelName: string
     }
-    advertisementFn: () => ServerAdvertisement
+    advertisementFn?: () => ServerAdvertisement
   }
 
   enum ClientStatus {
```
Comment (LucienHH) [inline index.d.ts]: Duplicate types

**Comment by LucienHH** [comment]
I don't think it's a good idea to add dependencies for eslint/prettier here even though they're in dev. PrismarineJS uses Standard JS across all repos and is best to keep it aligned that way.

**Comment by extremeheat** [comment]
Please revert the changes unrelated to the type fixes and we can merge.

**Comment by hvlxh** [comment]
@extremeheat

**Comment by extremeheat** [comment]
The only changed file in this PR should be index.d.ts

**Comment by hvlxh** [comment]
ah

**Comment by hvlxh** [comment]
@extremeheat you want any changes or someone should review it?

**Comment by extremeheat** [comment]
I'm afraid we can't accept the PR in the current state. Some of the functions exposed in the types (like toBase64) are not documented and are supposed to be internal functions. These shouldn't be in the types. 

There are also unrelated changes beyond the types (which are also missing comments and have deleted fields) that you would be better off freshly forking the bedrock-protocol master and manually applying your changes on a new branch, without using automated tools.

**Comment by hvlxh** [comment]
i will draft this ig

**Comment by hvlxh** [comment]
wait actually i will open an new one because new commits counts on old too

**Pull request closed without merging**